### ISSUE TITLE
timezone issue

### ISSUE THREAD

**Issue by NearestWater262** [uazo/bromite-buildtools#245]
I love that you have continued the bromite torch. The fingerprint measure are impeccable as well, better than brave and much more effective.

Can you provide an option to match time zone to IP? because otherwise it renders a VPN to be semi-useless. check out the Vytal extension for relavent code.

**Comment by uazo** [comment]
the issue is interesting but need to use external service (so privacy issue), because only an external server can indicate the device's outgoing IP

it seems to be possible to extract it from a dns call
```
dig +short myip.opendns.com @resolver1.opendns.com
```
some code in chromium:
- [IsVpnPresent](https://source.chromium.org/chromium/chromium/src/+/main:net/dns/dns_config_service_android.cc;l=40;bpv=1;bpt=1?q=IsVpnPresent&ss=chromium%2Fchromium%2Fsrc)
- [net/base/network_change_notifier.h](https://source.chromium.org/chromium/chromium/src/+/main:net/base/network_change_notifier.h;l=1;bpv=1;bpt=1)
- [P2PSocketManager::GetDefaultLocalAddress](https://source.chromium.org/chromium/chromium/src/+/main:services/network/p2p/socket_manager.cc;l=319;bpv=1;bpt=1?q=P2PSocketManager::GetDefaultLocalAddres&ss=chromium%2Fchromium%2Fsrc)

some ip to location list
- https://www.ip2location.com
- api https://www.ip2location.io/#ipl


but I would be interested to know how often device's IP changes and if so, to be notified so that can delete everything.

to be better evaluated

>  because otherwise it renders a VPN to be semi-useless.

I don't really agree, why do you say that?

> check out the Vytal extension for relavent code.

what is it? where can I find it?

**Comment by PF4Public** [comment]
@NearestWater262 Why do you insist on binding to IP? Won't random timezone do the trick?

**Comment by NearestWater262** [comment]
@PF4Public 
I guess, but the mismatching IP/timezone is a red flag and doesnt look "Normal". if you go on Fingerprint.com they talk about this also, how they use this to catch people they dont want avoiding fingerprints.

@uazo please see here
https://github.com/vytal-io/vytal-extension/blob/master/src/utils/getReverseGeocoding.ts
looks like they use openstreetmap

in the end, if you have to make an outgoing connection on startup, you could have it off by default, but it's so useful, it's definitely worth it. and Vytal has already written the code.

> why you say that?

not having this renders the VPN useless because if I am in Europe, and my phone is set to CET (European Central Time), but I have an Asian IP address, they are gonna at least guess i am in the CET time zone actually. And it really stands out as something unique: guy with Asian IP and Europe timezone

**Comment by uazo** [comment]
> is a red flag and doesnt look "Normal"

nothing in this browser can be defined as normal. :)
I wouldn't worry about the fake timezone...

> if you go on Fingerprint.com they talk about this also, how they use this to catch people they dont want avoiding fingerprints.

yes, I had read it. in fact, I have read so many documents on the subject, so many that my ideas are more confused than before.

> looks like they use openstreetmap

yes, i finally found it, but i still haven't read its privacy policy. have you? can you give us a summary?

> And it really stands out as something unique: guy with Asian IP and Europe timezone

hypothesis, where is the data to prove it? think of the opposite, if everyone uses vpn with chrome, the output ip of the vpn has everything but the real timezone of that area...
so far only google has them, and they are careful not to make them public.
although theoretically we could build them too... carl was firmly against it, but i'm thinking about it..

**Comment by NearestWater262** [comment]
@uazo okay so I made a mistake. They are using ip-api.com for the timezone info.

https://github.com/vytal-io/vytal-extension/blob/master/src/utils/getIp.ts

the privacy policy for them is actually really good and extremely simple. They don't keep a log of the client's IP after 1 min

https://ip-api.com/docs/legal

> hypothesis, where is the data to prove it? think of the opposite, if everyone uses vpn with chrome, the output ip of the vpn has everything but the real timezone of that area...

If someone uses an IP, and the time zone is different, then someone's general location can be inferred. It's not only weird/unique behavior, but it is also de-anonymizing.

**Comment by NearestWater262** [comment]
Yeah I know.. It's difficult :/ well, i think it should be an option still. It's a big data point. But you are the developer of course.
